### PR TITLE
report: add html report viewing support for safari

### PR DESCRIPTION
**Pull request by exterkamp**
**Summary**
Added Safari support!  Main areas of focus were:
- fixing tables
- fixing header sticky-scrolling
- fixing scroll > 100% making LH icon scale too large

### CODE REVIEW

File: lighthouse-core/report/html/report-styles.css [hunk 1]
```diff
@@ -930,14 +919,16 @@ summary.lh-passed-audits-summary {
 .lh-table-column--url {
   min-width: 250px;
   white-space: nowrap;
-  max-width: 0;
 }
 
-/* Keep bytes columns narrow if they follow the URL column */
+/* Keep columns narrow if they follow the URL column */
+/* 12% width was found to be a good width for en and i18n without being too wide */
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: > 12% was determined to be a decent narrow width, but wide enough for column headings

File: lighthouse-core/report/html/report-styles.css [hunk 2]
```diff
@@ -930,14 +919,16 @@ summary.lh-passed-audits-summary {
...
 .lh-table-column--url + th.lh-table-column--bytes,
 .lh-table-column--url + .lh-table-column--bytes + th.lh-table-column--bytes,
+.lh-table-column--url + .lh-table-column--ms, 
+.lh-table-column--url + .lh-table-column--ms + th.lh-table-column--bytes,
 .lh-table-column--url + .lh-table-column--bytes + th.lh-table-column--timespanMs {
-  max-width: var(--bytes-col-width);
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: you should delete this variable too.

File: lighthouse-core/report/html/templates.html [hunk 3]
```diff
@@ -83,7 +85,8 @@
 <template id="tmpl-lh-opportunity">
   <div class="lh-audit lh-audit--load-opportunity">
     <details class="lh-expandable-details">
-      <summary class="lh-audit__header lh-expandable-details__summary">
+        <summary>
+            <div class="lh-audit__header lh-expandable-details__summary">
         <div class="lh-load-opportunity__cols">
```
Comment (paulirish) [inline lighthouse-core/report/html/templates.html]: need some indentation fixes here

File: lighthouse-core/report/html/report-styles.css [hunk 4]
```diff
@@ -820,19 +821,6 @@
   line-height: var(--header-line-height);
 }
 
-.lh-passed-audits[open] summary.lh-passed-audits-summary {
-  margin-bottom: calc(var(--default-padding) * 2);
-}
-
-summary.lh-passed-audits-summary {
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: good catch that this is unused now.

File: lighthouse-core/report/html/report-styles.css [hunk 5]
```diff
@@ -930,14 +919,16 @@ summary.lh-passed-audits-summary {
...
 .lh-table-column--url + th.lh-table-column--bytes,
 .lh-table-column--url + .lh-table-column--bytes + th.lh-table-column--bytes,
+.lh-table-column--url + .lh-table-column--ms, 
+.lh-table-column--url + .lh-table-column--ms + th.lh-table-column--bytes,
 .lh-table-column--url + .lh-table-column--bytes + th.lh-table-column--timespanMs {
-  max-width: var(--bytes-col-width);
+  width: 12%;
 }
 
 .lh-table-column--code {
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css]: while you're here, i notice that this var doesnt exist anymore.

it's used on these two audits at least

image


seems like its fine without a max-width being set.

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 6]
```diff
@@ -218,8 +218,8 @@ class ReportUIFeatures {
   animateHeader() {
     const collapsedHeaderHeight = 50;
     const heightDiff = this.headerHeight - collapsedHeaderHeight + this.headerOverlap;
-    const scrollPct = Math.min(1,
-      this.latestKnownScrollY / (this.headerHeight - collapsedHeaderHeight));
+    const scrollPct = Math.max(0, Math.min(1,
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: i always did like devtools' `Number.prototype.constrain` method. :)

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (exterkamp) [reply lighthouse-core/report/html/report-styles.css]: Done!

File: lighthouse-core/report/html/renderer/report-ui-features.js (hunk 6 above)
Comment (patrickhulce) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: Funny story, I recently ran into a bizarre case where this pattern was eating up ~300ms of a ~2s execution in another codebase. I say this not to pass any comment on its usage here as it's definitely fine, but as a fun fact in my eternal struggle against @paulirish that perf 🚲 🏠 'ing can be educational and fun :D

File: lighthouse-core/report/html/report-styles.css [hunk 7]
```diff
@@ -648,6 +646,7 @@
 }
 
 .lh-header-sticky {
+  position: -webkit-sticky;
```
Comment (patrickhulce) [inline lighthouse-core/report/html/report-styles.css]: the time almost certainly isn't now, but do you see a path of us bringing in some build process with prefixers?

**Pull request merged**